rfc/graphql-schema/additionalTypeDefs/byPage/addresses.ts

```diff
@@ -0,0 +1,139 @@
+export default /* GraphQL */ `
+  type CompanyAddress {
+    entityId: Int!
+    firstName: String!
+    lastName: String!
+    address1: String!
+    address2: String
+    addressType: DestinationAddressType
+    city: String!
+    country: String
+    countryId: Int
+    countryCode: String!
+    stateOrProvince: String
+    stateId: Int
+    phone: String
+    postalCode: String
+    formFields: [CustomerFormFieldValue!]!
+
+    company: Company!
+
+    label: String!
+    isShipping: Boolean!
+    isBilling: Boolean!
+    isDefaultShipping: Boolean!
+    isDefaultBilling: Boolean!
+  }
+
+  type CompanyAddressEdge {
+    node: CompanyAddress!
+    cursor: String!
+  }
+
+  type CompanyAddressConnection {
+    edges: [CompanyAddressEdge!]!
+    pageInfo: PageInfo!
+    collectionInfo: CollectionInfo
+  }
+
+  input CompanyAddressFiltersInput {
+    search: String
+    city: String
+    state: String
+    country: String
+  }
+
+  extend type Company {
+    addresses(
+      filters: CompanyAddressFiltersInput
+      before: String
+      after: String
+      first: Int
+      last: Int
+    ): CompanyAddressConnection!
+  }
+
+  input CompanyAddressCreateInput {
+    firstName: String!
+    lastName: String!
+    address1: String!
+    address2: String
+    city: String!
+    countryCode: String!
+    stateOrProvince: String
+    phone: String
+    postalCode: String
+    extraFields: CustomerFormFieldsInput
+
+    label: String!
+    isShipping: Boolean!
+    isBilling: Boolean!
+    isDefaultShipping: Boolean!
+    isDefaultBilling: Boolean!
+  }
+
+  input CompanyAddressUpdateInput {
+    firstName: String
+    lastName: String
+    address1: String
+    address2: String
+    city: String
+    countryCode: String
+    stateOrProvince: String
+    phone: String
+    postalCode: String
+    extraFields: CustomerFormFieldsInput
+
+    label: String!
+    isShipping: Boolean
+    isBilling: Boolean
+    isDefaultShipping: Boolean
+    isDefaultBilling: Boolean
+  }
+
+  type CompanyAddressResult {
+    address: CompanyAddress
+    errors: [CompanyAddressError!]!
+  }
+
+  type CompanyAddressError implements Error {
+    message: String!
+  }
+
+  type CompanyAddressDeleteError implements Error {
+    message: String!
+  }
+
+  type CompanyAddressDeleteResult {
+    errors: [CompanyAddressDeleteError!]!
+  }
+
+  input SetAsDefaultInput {
+    isDefaultShipping: Boolean
+    isDefaultBilling: Boolean
+  }
+
+  type CompanyFormFields {
+    address: [FormField!]!
+  }
+
+  extend type FormFields {
+    company: CompanyFormFields!
+  }
+
+  extend type CompanyMutations {
+    addAddress(
+      companyId: ID!
+      input: CompanyAddressCreateInput
+    ): CompanyAddressResult!
+    updateAddress(
+      addressId: ID!
+      input: CompanyAddressUpdateInput!
+    ): CompanyAddressResult!
+    setAddressAsDefault(
+      addressId: ID!
+      input: SetAsDefaultInput!
+    ): CompanyAddressResult!
```

Comment:
seems like the customer update address operations, the `ID` is part of the input type, and the input type has an internal data type.

example:
```graphql
mutation MyMutation {
  customer {
    updateCustomerAddress(
      input: {addressEntityId: 123, data: {address1: "", address2: ""}}
    ) {
      address {
        address1
        address2
      }
    }
  }
}
```

might be good to just follow that pattern for simplicty sake?
and they also seem to all include `reCaptchaV2` as an input field - should we follow that?

cc: @snaderiBC do you or the customers team have context on the graphql api for customers?

Reply:
oh, captcha was an oversight, will add it. For the ID, I don't mind really, it looks similar, just using a single input 🤷

Reply:
I thought I'd seen examples of the ID being a sibling, but looking at the Storefront graph again they seem to be inlined into one argument, at least for updating addresses and cart items